service-base/src/main/java/org/eclipse/hono/service/HonoBeanNameGenerator.java

```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright (c) 2017 Bosch Software Innovations GmbH.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Bosch Software Innovations GmbH - initial creation
+ */
+
+package org.eclipse.hono.service;
+
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.beans.factory.support.BeanDefinitionRegistry;
+import org.springframework.beans.factory.support.BeanNameGenerator;
+
+/**
+ * Generates bean names that are identical to the class name.
+ */
+public class HonoBeanNameGenerator implements BeanNameGenerator {
```

Comment:
Regarding options 1 and 2 this was exactly my concern. For option 3 we could easily change `BeanNameGenerator` to respect names set by `@Component("myBean")`. 

Another option just came into my mind: how about a helper method that derives the bean name from the class name equivalent to Spring's `BeanNameGenerator`? This could be used like:
```
    @Bean
    public ObjectFactoryCreatingFactoryBean amqpServerFactory() {
        ObjectFactoryCreatingFactoryBean factory = new ObjectFactoryCreatingFactoryBean();
        factory.setTargetBeanName(HonoSpringUtils.decapitalize(DeviceRegistryAmqpServer.class.getSimpleName()));
        return factory;
    }
```
This had to be used everywhere one wants to derive the bean name by the class name and of course has again the problem that it would ignore bean names set explicitly in the `@Component` annotation.